pkgs/tools/text/sgrep/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, sgrep, fetchurl, runCommand, lib, m4, makeWrapper }:
+stdenv.mkDerivation rec {
+  pname = "sgrep";
+  version = "1.94a";
+
+  src = fetchurl {
+    url = "https://www.cs.helsinki.fi/pub/Software/Local/Sgrep/sgrep-${version}.tar.gz";
+    sha256 = "sha256-1bFkeOOrRHNeJCg9LYldLJyAE5yVIo3zvbKsRGOV+vk=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  preFixup = ''
+    wrapProgram $out/bin/sgrep \
```

Comment:
This should be done in postInstall. Also why do we need m4? That's really unusual.

Reply:
Applied.

> Also why do we need m4? That's really unusual.
sgrep shells out to m4 to pre-process expressions when the user wants to define and use custom macros.